src/qc_data_preparation/helpers/dataset.py

```diff
@@ -8,6 +9,36 @@
 from keras.utils.data_utils import get_file
 
 
+if sys.platform == "darwin":
+    import tempfile
+    from pathlib import PurePath
+    from kedro.io.core import get_filepath_str
+    from kedro_datasets.tensorflow import TensorFlowModelDataSet
+    from kedro_datasets.tensorflow.tensorflow_model_dataset import TEMPORARY_H5_FILE
+
+    def load_folder(self) -> tf.keras.Model:
+        load_path = get_filepath_str(self._get_load_path(), self._protocol)
+        print(f"load_path is {load_path}")
+
+        with tempfile.TemporaryDirectory(prefix=self._tmp_prefix) as path:
+            if self._is_h5:
+                path = str(PurePath(path) / TEMPORARY_H5_FILE)
+                self._fs.copy(load_path, path)
+            else:
+                self._fs.get(load_path + "/", path, recursive=True)
```

Comment:
IIRC this is the actual patch, it would be good to highlight this. Something like this:
```suggestion
            else:
                # MacOS Patch: add / to force copying the folder content only
                self._fs.get(load_path + "/", path, recursive=True)
```